core/capabilities/ccip/common/extradatacodec.go

```diff
@@ -10,15 +10,13 @@ import (
 
 // ExtraDataCodec is a struct that holds the chain specific extra data codec
 type ExtraDataCodec struct {
-	EVMExtraDataCodec    SourceChainExtraDataCodec
-	SolanaExtraDataCodec SourceChainExtraDataCodec
+	registeredExtraDataCodecMap map[string]SourceChainExtraDataCodec
```

Comment:
Isn't it little strange that we have 3-4 public types that are called `ExtraDataCodec`?

Also, there is no point in introducing variations like:

- `DecodeExtraArgs` and `SourceChainExtraDataCodec.DecodeExtraArgs`
- `DecodeTokenAmountDestExecData` and `SourceChainExtraDataCodec.DecodeDestExecDataToMap`

This is where a lot of unwarranted complexity comes from imo

Please try this:

- Rename as `ExtraDataCodecProvider` which can just be a map alias
- Add single fn as `Provide(selector cciptypes.ChainSelector) SourceChainExtraDataCodec`
- When you need a `SourceChainExtraDataCodec` for a chain (common type) use this provider

Reply:
This is used by a Dest Relayer, and it will handle incoming message/report from different source chain

Reply:
> * Rename as ExtraDataCodecProvider which can just be a map alias

This seems like a fine suggestion.

> * Add single fn as Provide(selector cciptypes.ChainSelector) SourceChainExtraDataCodec
> * When you need a SourceChainExtraDataCodec for a chain (common type) use this provider

This just seems different, I'm not sure it's better or worse.

Reply:
> I think returning an error when the chain selector doesn't have an implementation leads to more pragmatic code.

Yeah, for sure. My example was just quick pseudocode.